src/components/standup/index.tsx

```diff
@@ -0,0 +1,151 @@
+
+const StandUpContainer: FC = () => {
+    const [standupUpdate, setStandupUpdate] = useState<standupUpdateType>({
+        type: 'user',
+        completed: '',
+        planned: '',
+        blockers: '',
+    });
+
+    const [buttonDisable, setButtonDisable] = useState<boolean>(true);
+    const [addStandup] = useAddStandupMutation();
+    const { data: user } = useGetUserQuery();
+    const { data: userStandupdata } = useUserStandupDetailsQuery(user?.id);
+
+    const { SUCCESS, ERROR } = ToastTypes;
+    const standupDates = userStandupdata?.data?.map((element) => element.date);
+    const totalMissedUpdate = getTotalMissedUpdate(standupDates || []);
+    const yesterdayDate = moment().subtract(1, 'days').format('MMMM DD, YYYY');
+
+    const buttonStyleClass = buttonDisable
+        ? `${styles.nonActiveButton}`
+        : `${styles.activeButton}`;
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setStandupUpdate((prevStandupUpdate) => ({
+            ...prevStandupUpdate,
+            [event.target.name]: event.target.value,
+        }));
+    };
+    const isValidate = () => {
+        return (
+            standupUpdate.completed !== '' &&
+            standupUpdate.planned !== '' &&
+            standupUpdate.blockers !== ''
+        );
+    };
+    useEffect(() => {
+        const isValid = isValidate();
+        console.log(isValid);
+        setButtonDisable(!isValid);
+    }, [
+        standupUpdate.completed,
+        standupUpdate.planned,
+        standupUpdate.blockers,
+    ]);
+
+    const handleFormSubmission = async (
+        event: React.FormEvent<HTMLFormElement>
+    ) => {
+        event.preventDefault();
+        try {
+            await addStandup(standupUpdate);
+            toast(SUCCESS, 'Standup submitted successfully');
+            setStandupUpdate({
+                type: 'user',
+                completed: '',
+                planned: '',
+                blockers: '',
+            });
+        } catch (error) {
+            toast(ERROR, 'Something went wrong!');
+        }
+    };
+
+    return (
+        <>
+            <section className="container">
+                <div className={styles.standupContainer}>
+                    <div className={styles.standupBanner}>
+                        <p className={styles.bannerPara}>
+                            You have
+                            <span
+                                className={styles.totalMissedUpdate}
+                                data-testid="missed-updates"
+                            >
+                                {totalMissedUpdate} missed
+                            </span>
+                            Standup updates this week
+                        </p>
+                        <p className={styles.bannerPara}>
+                            Let&apos;s try to avoid having zero days{' '}
+                        </p>
+                    </div>
```

Comment:
This can be a separate component. Also, it has been created for the task progress you can reuse the component rather than re-writing.

Reply:
fixed